src/routes/estimations/estimations.controller.spec.ts

```diff
@@ -0,0 +1,288 @@
+import {
+  mockNetworkService,
+  TestNetworkModule,
+} from '../../datasources/network/__tests__/test.network.module';
+import { DomainModule } from '../../domain.module';
+import { chainBuilder } from '../../domain/chains/entities/__tests__/chain.builder';
+import { EstimationRequest } from '../../domain/estimations/entities/estimation-request.entity';
+import { estimationBuilder } from '../../domain/estimations/entities/__tests__/estimation.builder';
+import {
+  multisigTransactionBuilder,
+  toJson as multisigTransactionToJson,
+} from '../../domain/safe/entities/__tests__/multisig-transaction.builder';
+import { safeBuilder } from '../../domain/safe/entities/__tests__/safe.builder';
+import { DataSourceErrorFilter } from '../common/filters/data-source-error.filter';
+import { EstimationsModule } from './estimations.module';
+
+describe('Estimations Controller (Unit)', () => {
+  let app: INestApplication;
+  let safeConfigApiUrl: string;
+
+  beforeAll(async () => {
+    safeConfigApiUrl = faker.internet.url();
+    fakeConfigurationService.set('safeConfig.baseUri', safeConfigApiUrl);
+    fakeConfigurationService.set('exchange.baseUri', faker.internet.url());
+    fakeConfigurationService.set('exchange.apiKey', faker.datatype.uuid());
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    fakeCacheService.clear();
+
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [
+        // feature
+        EstimationsModule,
+        // common
+        DomainModule,
+        TestCacheModule,
+        TestConfigurationModule,
+        TestNetworkModule,
+      ],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    app.useGlobalFilters(new DataSourceErrorFilter());
+
+    await app.init();
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  describe('Create estimations', () => {
+    it('Success', async () => {
+      const chainId = faker.random.numeric();
+      const address = faker.finance.ethereumAddress();
+      const chain = chainBuilder().build();
+      const safe = safeBuilder().build();
+      const estimation = estimationBuilder().build();
+      const lastTransaction = multisigTransactionBuilder().build();
+      mockNetworkService.get.mockImplementation((url) => {
+        const chainsUrl = `${safeConfigApiUrl}/api/v1/chains/${chainId}`;
+        const getSafeUrl = `${chain.transactionService}/api/v1/safes/${address}`;
+        const multisigTransactionsUrl = `${chain.transactionService}/api/v1/safes/${address}/multisig-transactions/`;
+        if (url === chainsUrl) {
+          return Promise.resolve({ data: chain });
+        }
+        if (url === getSafeUrl) {
+          return Promise.resolve({ data: safe });
+        }
+        if (url === multisigTransactionsUrl) {
+          return Promise.resolve({
+            data: {
+              count: 1,
+              results: [multisigTransactionToJson(lastTransaction)],
+            },
```

Comment:
An alternative would be something like:

```typescript
pageBuilder()
  .with('count', 1)
  .with('results', [multisigTransactionToJson(lastTransaction)])
  .build();
```

No strong opinion on this one. I'll leave it up to you 👍

Reply:
Yes, I think the way you proposed it's more idiomatic. I'll change it 👍🏻

Reply:
Changed in [cc1b105](https://github.com/5afe/safe-client-gateway-nest/pull/269/commits/cc1b10550ff69e6516e4a0956cc3820a7d6f5c43)